ssl/statem/statem_srvr.c

```diff
@@ -1492,6 +1496,84 @@ MSG_PROCESS_RETURN tls_process_client_hello(SSL_CONNECTION *s, PACKET *pkt)
     static const unsigned char null_compression = 0;
     CLIENTHELLO_MSG *clienthello = NULL;
 
+#ifndef OPENSSL_NO_ECH
+    /*
+     * For split-mode we want to have a way to point at the CH octets
+     * for the accept-confirmation calculation.
+     */
+    if (s == NULL)
+        return 0;
+    if (pkt == NULL) {
+        SSLfatal(s, SSL_AD_INTERNAL_ERROR, ERR_R_INTERNAL_ERROR);
+        goto err;
+    }
+    if (s->server == 1 && PACKET_remaining(pkt) != 0) {
```

Comment:
Isn't `s->server` always 1 here? `tls_process_client_hello` is a server side only function?

Reply:
`s->server` is effectively a boolean so I'd prefer just `if (s->server && ...`

Reply:
sure